Samira-React-w1-assignment

samira313 opened the pull request:
I built a small e-commerce React app using Vite. The app displays a list of products and allows the user to filter them by category. I used useState to manage the selected category and applied dynamic filtering. I styled the components with CSS and deployed the app to Netlify by setting the correct base and publish directories. Now it’s live and accessible online!
https://euphonious-wisp-2a91f6.netlify.app/

tkitson reviewed (changes requested):
Great work @samira313 - the functionality is pretty much there, I've suggested a couple of extra tweaks. Nice work making it responsive, the css grid is a good way to do this. Let me know here or in Slack if you have any questions!

samira313 commented:
Hi Tom!


Thanks for your helpful feedback.

I’ve moved the logic and layout into a separate Home.jsx page under the
pages/ folder as you suggested, and I’ve added the logic to deselect a
category when clicking the active one — using a ternary with the
selectedCategory state.

Let me know if there’s anything else you’d like me to improve!


Thanks again!

Samira Ahmadi


On Mon, 21 Apr 2025 at 18:13, Tom Kitson ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Great work @samira313 <https://github.com/samira313> - the functionality
> is pretty much there, I've suggested a couple of extra tweaks. Nice work
> making it responsive, the css grid is a good way to do this. Let me know
> here or in Slack if you have any questions!
> ------------------------------
>
> In week1/project/ecommerce/src/components/CategoryList.jsx
> <https://github.com/HackYourAssignment/React-cohort51/pull/3#discussion_r2052650047>
> :
>
> > @@ -0,0 +1,22 @@
> +import React from "react";
> +import "../styles/CategoryList.css";
> +
> +function CategoryList({ categories, selectedCategory, onSelectCategory }) {
> +  return (
> +    <div className="category-list">
> +      {categories.map((category, index) => (
> +        <button
> +          key={index}
> +          onClick={() => onSelectCategory(category)}
>
> Nice work @samira313 <https://github.com/samira313> ! You can add some
> logic to the onSelectCategory function to deselect a category, if it's
> currently selected and clicked again. Can you see if you can add this? Try
> using a ternary statement, and the selectedCategory variable
> ------------------------------
>
> On week1/project/ecommerce/src/App.jsx
> <https://github.com/HackYourAssignment/React-cohort51/pull/3#discussion_r2052660491>
> :
>
> I have a structural change suggestion: rather than rendering all of your
> components in the App.jsx file, try to move them into a new component
> inside a folder called 'pages'. You could in fact take your ProductList.jsx
> component and use this - just add the Header and CategoryList, and this is
> basically your entire page.
>
> I'm just suggesting this as it's good practice for a React app - having
> separate components for pages - rather than rendering everything inside the
> App.jsx file. Let me know here or on Slack if you have any questions!
> Tom
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/HackYourAssignment/React-cohort51/pull/3#pullrequestreview-2781667898>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKWT54L3AYJGEWB6MESDB2L22UKK7AVCNFSM6AAAAAB3HUWL32VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDOOBRGY3DOOBZHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
